File: src/main/java/gargoyle/ct/LocaleProvider.java
package gargoyle.ct;

import java.util.Locale;

public class LocaleProvider {

    private Locale locale;

    public LocaleProvider() {
        locale = Locale.getDefault();
    }

    public Locale getLocale() {
        return locale;
    }

    public void setLocale(Locale locale) {
        this.locale = locale;
    }
}

File: util/src/main/kotlin/gargoyle/ct/util/messages/LocaleProvider.kt
package gargoyle.ct.util.messages

import gargoyle.ct.util.prop.CTObservableProperty
import java.util.*

interface LocaleProvider {
    fun getLocale(): Locale
    fun setLocale(locale: Locale)
    fun locale(): CTObservableProperty<SupportedLocales>
}
